lib/IRGen/GenObjC.cpp

```diff
@@ -363,6 +363,10 @@ IRGenModule::getObjCProtocolGlobalVars(ProtocolDecl *proto) {
   protocolLabel->setSection(GetObjCSectionName("__objc_protolist",
                                                "coalesced,no_dead_strip"));
 
+  // Add __objc_protolist into llvm.used to avoid runtime crash.
```

Comment:
@jinlin-bayarea might help to add some information regarding the crash, perhaps LTO?

Reply:
I think that a comment along the lines of:
```
Mark used to prevent DCE of public unreferenced protocols to ensure that they are available for external use when used a module is used as a library.
```
should be good.